src/api/dependency-graph/src/graph-collector/index.js

```diff
@@ -0,0 +1,136 @@
+const yaml = require('js-yaml');
+const fs = require('fs');
+const { logger, fetch } = require('@senecacdot/satellite');
+const Graph = require('./graph');
+
+function extractRepositoryUrl(repository) {
```

Comment:
Instead of doing this, why don't we use the package name you already have from `pnpm` to query the npmjs registry programmatically?  For example, you can get the package info for React using https://registry.npmjs.org/react.  This will return everything we need in JSON format.

Here are some docs:

- https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md
- https://www.edoardoscibona.com/exploring-the-npm-registry-api
- https://www.npmjs.com/package/query-registry
- https://www.jsdocs.io/package/query-registry